### PR TITLE
{Doc} Update cffi troubleshooting, extension authoring and onboarding guide

### PR DESCRIPTION
**Pull request by fengzhou-msft**
**Description**  

Update cffi troubleshooting when there's no RPM package;
Add back vendored SDK guidance;
Add PLR process in onboarding guide.

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Doc

File: doc/onboarding_guide.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
 Onboarding Best Practices
 =========================
 
-Reach out to `azpycli@microsoft.com` to get the process started. You'll be assigned a dev contact on the CLI team. Early and frequent communication with this contact is essential to ensuring a smooth onboarding.
+As a prerequisite, please contact Carl Shipley first for the [PLR](http://aka.ms/plrcriteria) (Product Launch Readiness) process.
+
+Then reach out to `azpycli@microsoft.com` to get the CLI onboarding process started. You'll be assigned a dev contact on the CLI team. Early and frequent communication with this contact is essential to ensuring a smooth onboarding.
```
Comment (qwordy) [inline doc/onboarding_guide.md:6]: ensuring -> ensure?

Comment (fengzhou-msft) [reply doc/onboarding_guide.md:6]: It looks like both usage are OK. In the case of `ensuring`, the `ensuring a smooth onboarding` part is called a `gerund phrase`, which functions as nouns.

Comment (qwordy) [reply doc/onboarding_guide.md:6]: Professional!

**Review by qwordy — APPROVED** [review]

**Pull request merged**